### DISCUSSION TITLE
Microsoft owns Github

### DISCUSSION THREAD

**Discussion by grendel151** [valinet/ExplorerPatcher#3734] [General]
First, thank you all for creating and making this program available.  What MS is doing, especially with false flagging things as a virus is disgusting.  Your work on helping the OS run like people want it to is greatly appreciated.

But, with that said...

Have y'all thought about looking for a new home for this project?  You kinda live on their turf here on Github.  I wouldn't be surprised if your time here is limited and I'm guessing you won't get much of a warning.

**Comment by Amrsatrio** [comment]
Yes. A self-hosted Git server. Once that happens of course.

**Comment by wuyilingwei** [comment]
You're right, this is a problem, but I'm glad I haven't seen MS abuse their control over github/bing, at least I haven't heard of it.

**Comment by mistertheloaf** [comment]
Explorer Patcher is a really great program.  I can't stand to use any computer without it.   But, I do not believe that "Microsoft hates Explorer Patcher".

The problem is, Explorer is a core component of Windows and Microsoft doesn't want people messing with it because they (Microsoft) don't want to get complaints that Windows isn't working properly.  And that has actually happened.

Raymond Chen, who works for Microsoft, [wrote about it on his blog last year](url).

> I did some bulk analysis of all the crash dumps that we received and observed that one third-party DLL was common to all of them.  This program patches Explorer in order to accomplish its enhancements.
>
> We recently made changes to this internal function .... whenever a new build goes out there’s a spike of Explorer crashes because all of these patchers start patching the wrong code, and Explorer starts crashing

Obviously, it would be really nice if Microsoft would stop screwing up Windows so that programs like EP aren't necessary in the first place.

**Reply by pyrates999** [reply]
Raymond Chen link you linked to is 404

**Reply by Amrsatrio** [reply]
If not hatred then I demand an explanation from them or a communication channel for discussing about this. Why they are doing this, and how we can get out of this. The problem is those have not yet been realized at all.

And I get your opinion. EP only exists because of their questionable changes. EP is here to make stuff usable like they used to be. Unfortunately it has to be as invasive and as unsupported to achieve its goals. Just recently they have been nuking old stuff from explorer.exe and look at the effort we had to make, and in result a price to pay for end users.

I did what I had to do, if this ended up being the reality, so be it.
